[docs] Remove Expokit pages from archive and related references from glossary (#21345)
expo/expo: docs/pages/archive/glossary.mdx

---
title: Glossary of terms
---

> > **error** The following page consists of all deprecated technologies that no longer exist.

Reviewer: ```suggestion
> **error** The following page consists of definitions for some of the deprecated terms, technologies and tools.
```

Phrasing nit.
